tiledb/sm/rest/curl.cc

```diff
@@ -451,6 +452,53 @@ Status Curl::make_curl_request(
       static_cast<void*>(&cb));
 }
 
+/**
+ * Provides crucial information on core-to-REST-server HTTP operations.  This is
+ * essential for analyzing and minimizing remote-request latencies.  An
+ * indispensable counterpart to Jaeger tracing, while Jaeger tracing isn't
+ * enough to give us a full picture on all interactions in all contexts.
+ *
+ * Easiest enable:
+ *     export TILEDB_REST_TRACE_CURL_CALLS=true
+ *     export TILEDB_CONFIG_LOGGING_LEVEL=5
+ * (you need both)
+ */
+CURLcode Curl::curl_maybe_instrumented(
+    const char* const url, uint8_t retry_number) const {
+  CURL* curl = curl_.get();
+  bool rest_trace_curl_calls = false;
+  bool found = false;
+  auto ok =
+      config_
+          ->get<bool>("rest.trace.curl.calls", &rest_trace_curl_calls, &found)
+          .ok();
+  if (!ok || !found || !rest_trace_curl_calls) {
+    return curl_easy_perform(curl);
+  }
+
+  uint64_t t1 = tiledb::sm::utils::time::timestamp_now_ms();
+  CURLcode curl_code = curl_easy_perform(curl);
+  uint64_t t2 = tiledb::sm::utils::time::timestamp_now_ms();
+  uint64_t dt = t2 - t1;
+  long http_code = 0;
+  if (curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code) != CURLE_OK) {
```

Comment:
A note to myself, that in case of tracing enabled we will be calling `curl_easy_getinfo` twice , since we also call it in `should_retry` : consider merging in `should_retry` to avoid that